### PR TITLE
AEA-976 remove command: support dependencies removal

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

cli remove command supports dependencies removal
## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/upgrade.py [hunk 1]
```diff
@@ -245,13 +126,23 @@ def upgrade_project(ctx: Context) -> None:  # pylint: disable=unused-argument
...
         click.echo("Shared dependencies removed.")
 
     for upgrader in upgraders:
         upgrader.remove_item()
-        upgrader.remove_dependencies_for_item()
         upgrader.add_item()
+
+    click.echo(
+        f"111 {ctx.agent_config.protocols} {shared_deps_to_remove} {shared_deps}"
+    )
```
Comment (DavidMinarsch) [inline aea/cli/upgrade.py:144]: Remove this .

File: aea/cli/upgrade.py [hunk 2]
```diff
@@ -358,18 +250,27 @@ def check_upgrade_is_required(self) -> str:
...
-        add_item(self.ctx, str(self.item_type), self.item_public_id)
+        current_item = get_item_public_id_by_author_name(
+            self.ctx.agent_config,
+            self.item_type,
+            self.item_public_id.author,
+            self.item_public_id.name,
+        )
+        if not current_item:
+            click.echo(f"Adding item { self.item_type} {self.item_public_id}.")
+            add_item(self.ctx, str(self.item_type), self.item_public_id)
```
Comment (DavidMinarsch) [inline aea/cli/upgrade.py:273]: why has this changed?

File: aea/cli/upgrade.py [hunk 3]
```diff
@@ -358,18 +250,27 @@ def check_upgrade_is_required(self) -> str:
...
-            )
+        remove_item = RemoveItem(
+            self.ctx,
+            self.item_type,
+            self.item_public_id,
+            with_dependencies=True,
+            force=True,
+        )
+        remove_item.remove()
+        click.echo(f"Item { self.item_type} {self.item_public_id} removed!")
```
Comment (DavidMinarsch) [inline aea/cli/upgrade.py:261]: remove unnecessary whitespace

File: aea/cli/upgrade.py (hunk 2 above)
Comment (DavidMinarsch) [reply aea/cli/upgrade.py:273]: (also, remove unnecessary whitespace in click echo)

File: aea/cli/upgrade.py (hunk 1 above)
Comment (solarw) [reply aea/cli/upgrade.py:144]: sorry, will do

File: aea/cli/upgrade.py (hunk 2 above)
Comment (solarw) [reply aea/cli/upgrade.py:269]: going to remove after some checks. suppose it will be unused

File: aea/cli/upgrade.py (hunk 3 above)
Comment (solarw) [reply aea/cli/upgrade.py:257]: done

File: aea/cli/upgrade.py (hunk 2 above)
Comment (solarw) [reply aea/cli/upgrade.py]: done

File: aea/cli/remove.py [hunk 4]
```diff
@@ -87,65 +105,330 @@ def skill(ctx: Context, skill_id):
...
+        Remove item.
+
+        Removed from the filesystem.
+        Removed from the agent configuration
+
+        Does not remove dependencies, please use `remove_dependencies`.
+        """
+        if (not self.force) and self.is_required_by:
+            raise click.ClickException(
+                f"Package {self.item_type} {self.item_id} can not be removed cause required by {','.join(map(str, self.required_by))}"
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:328]:
```suggestion
                f"Package {self.item_type} {self.item_id} can not be removed because it is required by {','.join(map(str, self.required_by))}"
```

File: aea/cli/remove.py [hunk 5]
```diff
@@ -16,27 +16,45 @@
...
+    DEFAULT_AEA_CONFIG_FILE,
+    PackageConfiguration,
+    PackageId,
+    PublicId,
+)
 
 
 @click.group()
+@click.option(
+    "--with-dependencies",
```
Comment (DavidMinarsch) [inline aea/cli/remove.py:47]: also allow abbreviated version `"-w"`

**Review by DavidMinarsch — APPROVED** [review]
LGTM, some minor comments

File: aea/cli/remove.py (hunk 5 above)
Comment (solarw) [reply aea/cli/remove.py:47]: done

File: aea/cli/remove.py [hunk 6]
```diff
@@ -87,65 +105,330 @@ def skill(ctx: Context, skill_id):
...
+            RemoveItem(
+                self.ctx,
+                str(dependency.package_type),
+                dependency.public_id,
+                with_dependencies=False,
+                force=True,
+            ).remove_item()
+            click.echo(
+                f"{dependency.package_type} {dependency.public_id} was removed from {self.agent_name}..."
+            )
```
[the remark this answers is not in the document]
Comment (solarw) [reply aea/cli/remove.py:418]: done

File: aea/cli/remove.py (hunk 4 above)
Comment (solarw) [reply aea/cli/remove.py:328]: done
